runtime/vm/vmprops.c

```diff
@@ -530,6 +530,103 @@ getLibSubDir(J9JavaVM *vm, const char *subDir, char **value)
 }
 
 #define COM_SUN_MANAGEMENT "com.sun.management"
+
+UDATA
+loadCMDLineSystemProperties(J9JavaVM *vm, JavaVMInitArgs *initArgs)
+{
+	UDATA rc = J9SYSPROP_ERROR_NONE;
+	BOOLEAN addManagementModule = FALSE;
+	jint i = 0;
+	const char* propValue = NULL;
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
+	/* Create the -D properties. This may override any of the writeable properties above.
+	 * Should the command line override read-only props?
+	 */
+	for (i = 0; i < initArgs->nOptions; ++i) {
+		char * optionString = initArgs->options[i].optionString;
+
+		if (0 == strncmp("-D", optionString, 2)) {
+			J9VMSystemProperty *currentProp = NULL;
+			char *propNameCopy = NULL;
+			char *propValueCopy = NULL;
+			UDATA propNameLen = 0;
+
+			propValue = strchr(optionString + 2, '=');
+			if (NULL == propValue) {
+				propNameLen = strlen(optionString) - 2;
+				propValue = optionString + 2 + propNameLen;
+			} else {
+				propNameLen = propValue - (optionString + 2);
+				++propValue;
+			}
+
+			{
+				UDATA valueLength = strlen(propValue);
+
+				propNameCopy = (char *)getMUtf8String(vm, optionString + 2, propNameLen); /* get a copy of the property name */
+				if (NULL == propNameCopy) {
+					rc = J9SYSPROP_ERROR_OUT_OF_MEMORY;
+					goto fail;
+				}
+				propValueCopy = (char *)getMUtf8String(vm, propValue, valueLength); /* get a copy of the property value */
```

Comment:
Similar comment for the comment.
In addition, all `getMUtf8String` usages were cast to `char *`, it can be refactored to return `char *` instead.

Reply:
all chars are treated as U_8 in this file, if it matters I'd rather make this change in a separate PR

Reply:
No concern here, I am fine either way.